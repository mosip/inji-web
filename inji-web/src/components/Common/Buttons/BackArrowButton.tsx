import React from 'react';
import BackArrowIcon from '../../../assets/BackArrowIcon.svg';

interface BackArrowButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const BackArrowButton: React.FC<BackArrowButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-back-arrow-container',
    iconTestId = 'icon-back-arrow',
    alt = 'Back Arrow'
}) => {
    return (
        <button
            type="button"
            onClick={onClick}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
