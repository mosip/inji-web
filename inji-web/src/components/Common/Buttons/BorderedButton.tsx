import React from "react";
import {renderGradientText} from "../../../utils/builder";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={props.fullWidth ? "w-full" : ""}>
    <div className={"bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"} onClick={props.onClick}>
            <div className={`py-1 px-2 rounded-lg bg-white justify-center center ${props.className}`}>
                <button
                    data-testid={props.testId}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title && renderGradientText(props.title)}
                </button>
            </div>
    </div>
  </div>
}

export type BorderedButtonProps = {
    fullWidth?: boolean | false;
    testId: string;
    onClick: (event: React.MouseEvent)=> void;
    title?: string;
    className?: string;
}
