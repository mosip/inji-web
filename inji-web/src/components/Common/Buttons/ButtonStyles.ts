export const ButtonStyles = {
    responsiveButtonWithText: {
        container: "flex items-center space-x-2 w-full",
        smallDeviceView: " w-full",
        smallDeviceButton: "sm:hidden",
        largeDeviceContainer: "hidden sm:flex items-center space-x-2",
        gradientContainer: "rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary p-[2px] transition-all duration-200 ease-in-out",
        iconButton: "flex items-center justify-center w-10 h-10 rounded-xl shadow border-none bg-[#FBF2EF] transition-all duration-200 ease-in-out",
        hoveredIconButton: "bg-gradient-to-r from-iw-primary to-iw-secondary"
    }
}