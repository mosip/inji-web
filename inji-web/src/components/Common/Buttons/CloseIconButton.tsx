import React from 'react';
import BackArrowIcon from '../../../assets/CloseIcon.svg';

interface CloseIconButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const CloseIconButton: React.FC<CloseIconButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-close-container',
    iconTestId = 'icon-close',
    alt = "Close"
}) => {
    return (
        <button
            type="button"
            onClick={(e)=>{
                e.stopPropagation()
                onClick()
            }}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
