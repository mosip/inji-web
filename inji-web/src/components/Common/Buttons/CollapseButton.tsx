import { useSelector } from "react-redux";
import { isRTL } from "../../../utils/i18n";
import { RootState } from "../../../types/redux";
import CollapseIcon from '../../../assets/CollapseIcon.svg';

type CollapseButtonProps = {
    isCollapsed: boolean;
    onClick: () => void;
    className?: string;
};

export const CollapseButton: React.FC<CollapseButtonProps> = ({isCollapsed, onClick, className}) => {
    const language = useSelector((state: RootState) => state.common.language);

    return (
        <button
            onClick={onClick}
            className={className}
        >
            <img
                src={CollapseIcon}
                alt="Collapse"
                className={`transform ${
                    isCollapsed
                        ? isRTL(language)
                            ? ''
                            : 'rotate-180'
                        : isRTL(language)
                        ? 'rotate-180'
                        : ''
                } transition-transform duration-300 min-w-[26px] min-h-[26px]`}
            />
        </button>
    );
};