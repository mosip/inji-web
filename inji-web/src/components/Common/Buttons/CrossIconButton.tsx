import React from 'react';
import CrossIcon from '../../../assets/CrossIcon.svg';

interface CrossIconButtonProps {
  onClick: () => void;
  btnClassName?: string;
  iconClassName?: string;
  btnTestId: string;
  iconTestId: string;
}

export const CrossIconButton: React.FC<CrossIconButtonProps> = ({
  onClick,
  btnClassName = "",
  iconClassName= "",
  btnTestId = "btn-close-icon-container",
  iconTestId = "icon-close",
}) => {
  return (
    <button
      type="button"
      onClick={onClick}
      data-testid={btnTestId}
      className={btnClassName}
    >
      <img
        src={CrossIcon}
        alt="Close"
        data-testid={iconTestId}
        className={iconClassName}
      />
    </button>
  );
};