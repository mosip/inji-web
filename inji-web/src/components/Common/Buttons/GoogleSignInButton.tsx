import { useState } from "react";
import { FcGoogle } from "react-icons/fc";

export const GoogleSignInButton:React.FC<GoogleSignInButtonProps> = (props) => {
  const [isLoading, setIsLoading] = useState(false);
  const onClickHandler = () => {
    setIsLoading(true);
    props.handleGoogleLogin();
  };

  return (
    <button
      onClick={onClickHandler}
      disabled={isLoading}
      data-testid="google-login-button"
      className="w-full bg-white flex items-center justify-center gap-2.5 break-words py-2 px-4 shadow-md border border-gray-300 rounded-xl"
    >
      <FcGoogle size={24} className="flex-shrink-0" />
      {isLoading ? props.loadingText : props.text}
    </button>
  );
};

export type GoogleSignInButtonProps = {
    handleGoogleLogin: () => void;
    loadingText: string;
    text:string
  };