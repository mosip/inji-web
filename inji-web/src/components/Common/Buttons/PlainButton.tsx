import React from "react";
import {renderGradientText} from "../../../utils/builder";

export const PlainButton:React.FC<PlainButtonProps> = (props) => {
    return <div className={props.fullWidth ? "w-full" : ""}>
        <div className={"rounded-lg text-center cursor-pointer"} onClick={props.onClick}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    className="font-bold rounded-lg">
                    {props.disableGradient ? props.title : renderGradientText(props.title)}
                </button>
            </div>
        </div>
    </div>
}

export type PlainButtonProps = {
    fullWidth?: boolean | false;
    testId: string;
    onClick: ()=> void;
    title: string;
    disableGradient?: boolean; 
}
