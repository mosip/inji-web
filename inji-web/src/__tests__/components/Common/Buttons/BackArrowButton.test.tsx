import {render, screen, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import {BackArrowButton} from '../../../../components/Common/Buttons/BackArrowButton';

jest.mock(
    '../../../../assets/BackArrowIcon.svg',
    () => 'mocked-back-arrow-path'
);

describe('BackArrowButton Component', () => {
    const mockOnClick = jest.fn();

    const defaultProps = {
        onClick: mockOnClick,
        btnTestId: 'btn-back-arrow-container',
        iconTestId: 'icon-back-arrow',
        alt: 'Back Arrow'
    };

    const renderBackArrowButton = (props = {}) => {
        return render(<BackArrowButton {...defaultProps} {...props} />);
    };

    beforeEach(() => {
        mockOnClick.mockClear();
    });

    test('matches snapshot with default props', () => {
        const {asFragment} = renderBackArrowButton();

        expect(asFragment()).toMatchSnapshot();
    });

    test('matches snapshot with custom props', () => {
        const {asFragment} = renderBackArrowButton({
            btnClassName: 'custom-btn-class',
            iconClassName: 'custom-icon-class',
            btnTestId: 'custom-btn-test-id',
            iconTestId: 'custom-icon-test-id',
            alt: 'Custom Alt Text'
        });

        expect(asFragment()).toMatchSnapshot();
    });

    test('renders button with provided btnTestId', () => {
        const customBtnTestId = 'test-btn-id';
        
        renderBackArrowButton({btnTestId: customBtnTestId});
        
        expect(screen.getByTestId(customBtnTestId)).toBeInTheDocument();
    });

    test('renders icon with provided iconTestId', () => {
        const customIconTestId = 'test-icon-id';
        
        renderBackArrowButton({iconTestId: customIconTestId});
        
        expect(screen.getByTestId(customIconTestId)).toBeInTheDocument();
    });

    test('renders with default test IDs when not provided', () => {
        renderBackArrowButton();
        
        expect(screen.getByTestId(defaultProps.btnTestId)).toBeInTheDocument();
        
        expect(screen.getByTestId(defaultProps.iconTestId)).toBeInTheDocument();
    });

    test('calls onClick handler when button is clicked', () => {
        renderBackArrowButton();
        
        fireEvent.click(screen.getByTestId(defaultProps.btnTestId));
        
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    test('applies custom button classes', () => {
        const customBtnClass = 'test-custom-class';
        renderBackArrowButton({btnClassName: customBtnClass});
        
        const button = screen.getByTestId(defaultProps.btnTestId);
        
        expect(button).toHaveClass(customBtnClass);
    });

    test('applies custom icon classes', () => {
        const customIconClass = 'test-icon-class';
        renderBackArrowButton({iconClassName: customIconClass});
        
        const img = screen.getByTestId(defaultProps.iconTestId);
        
        expect(img).toHaveClass(customIconClass);
    });

    test('renders the back arrow icon image with correct src', () => {
        renderBackArrowButton();
        const imgElement = screen.getByAltText(defaultProps.alt);
        
        expect(imgElement).toBeInTheDocument();
        
        expect(imgElement).toHaveAttribute('src', 'mocked-back-arrow-path');
    });

    test('applies default empty string as class when no className provided', () => {
        renderBackArrowButton();
        
        const button = screen.getByTestId(defaultProps.btnTestId);
        const img = screen.getByTestId(defaultProps.iconTestId);

        expect(button.className).toBe('');
        expect(img.className).toBe('');
    });

    test('sets custom alt text when provided', () => {
        const customAltText = 'Custom Alt Text Provided';
        renderBackArrowButton({alt: customAltText});
        
        const imgElement = screen.getByAltText(customAltText);
        
        expect(imgElement).toBeInTheDocument();
    });
});
