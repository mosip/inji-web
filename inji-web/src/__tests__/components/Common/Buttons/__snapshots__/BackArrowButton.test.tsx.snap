// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackArrowButton Component matches snapshot with custom props 1`] = `
<DocumentFragment>
  <button
    class="custom-btn-class"
    data-testid="custom-btn-test-id"
    type="button"
  >
    <img
      alt="Custom Alt Text"
      class="custom-icon-class"
      data-testid="custom-icon-test-id"
      src="mocked-back-arrow-path"
    />
  </button>
</DocumentFragment>
`;

exports[`BackArrowButton Component matches snapshot with default props 1`] = `
<DocumentFragment>
  <button
    class=""
    data-testid="btn-back-arrow-container"
    type="button"
  >
    <img
      alt="Back Arrow"
      class=""
      data-testid="icon-back-arrow"
      src="mocked-back-arrow-path"
    />
  </button>
</DocumentFragment>
`;
